### ISSUE TITLE
Parser produces wrong projection data type when parsing doubly nested derived table

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#14373]
Related to https://github.com/jOOQ/jOOQ/issues/14372, this test fails:

```java
@Test
public void testUnqualifiedFieldFromNestedDerivedTables() {
    withMeta("create table t (i int); create table u (j int)");

    assertProjectionEqualsAll(
        asList(field(unquotedName("y", "i"), INTEGER)),
        """
        select y.i from (
          select t.i from t
        ) y
        """, // This query works
        """
        select y.i from (
          select x.i from (
            select t.i from t
          ) x
        ) y
        """ // This query fails
    );
}
```

The failure is:

```
java.lang.AssertionError: Data type at position 0 expected:<integer /* java.lang.Integer */> but was:<other /* java.lang.Object */>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:120)
	at org.jooq.test.util.BaseTest.assertEquals(BaseTest.java:299)
	at org.jooq.impl.ParserMetaTest.assertProjectionEquals(ParserMetaTest.java:1326)
	at org.jooq.impl.ParserMetaTest.assertProjectionEqualsAll(ParserMetaTest.java:1333)
	at org.jooq.impl.ParserMetaTest.testUnqualifiedFieldFromNestedDerivedTables(ParserMetaTest.java:572)
```

**Comment by lukaseder** [comment]
This is a regression introduced by https://github.com/jOOQ/jOOQ/issues/14343

**Comment by lukaseder** [comment]
Luckily, #14343 hasn't been released yet, so only `SNAPSHOT` versions have been affected.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0, 3.17.6 (https://github.com/jOOQ/jOOQ/issues/14374), and 3.16.12 (https://github.com/jOOQ/jOOQ/issues/14375)
